specification/vmwarevirtustream/resource-manager/Microsoft.VMwareVirtustream/preview/2016-09-01-preview/vmwarevirtustream.json

```diff
@@ -0,0 +1,1304 @@
+        "x-ms-long-running-operation": true,
+        "tags": [
+          "Clusters"
+        ],
+        "summary": "modify cluster",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "name": "privateCloudName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the private cloud."
+          },
+          {
+            "name": "clusterName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the cluster."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ClusterRequest"
+            }
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/ClusterResponse"
+            }
+          },
+          "202": {
+            "description": "accepted operation"
+          }
+        },
+        "x-ms-examples": {
+          "Clusters_Update": {
+            "$ref": "./examples/Clusters_Update.json"
+          }
+        }
+      },
+      "delete": {
+        "operationId": "Clusters_Delete",
+        "x-ms-long-running-operation": true,
+        "tags": [
+          "Clusters"
+        ],
+        "summary": "delete cluster",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "name": "privateCloudName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the private cloud."
+          },
+          {
+            "name": "clusterName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the cluster."
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "accepted operation"
+          },
+          "204": {
+            "description": "no content"
+          }
+        },
+        "x-ms-examples": {
+          "Clusters_Delete": {
+            "$ref": "./examples/Clusters_Delete.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.VMwareVirtustream/privateClouds/{privateCloudName}/addAuthorization/{authorizationName}": {
```

Comment:
Standard guidance would be to either implement this as a POST action directly on the private cloud resource (i.e. `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.VMwareVirtustream/privateClouds/{privateCloudName}/addAuthorization` where the body includes the {authorizationName}), or to implement "authorizations" as an actual proxy resource and support CRUD on it - ARM is quite specific about how it interprets key-value pairs in resource URIs.

Reply:
URL not correct for the POST. What is this supposed to do?

Reply:
As per these comments ^ the URL shouldn't include the `/{authorizationName}` at the end, or ARM will misinterpret this action. https://armwiki.azurewebsites.net/faq/armbasics.html has some information.